pkg/cache/v3/simple.go

```diff
@@ -306,6 +306,39 @@ func (cache *snapshotCache) CreateWatch(request *Request, value chan Response) f
 	snapshot, exists := cache.snapshots[nodeID]
 	version := snapshot.GetVersion(request.TypeUrl)
 
+	if exists && knownResourceNames != nil {
```

Comment:
If this is the initial request and theres no previously known resources will this be guaranteed to be initialized?

Reply:
Not sure if I understand the question. `knownResourceNames` is initialized when a new stream is created in sowt/server.go Line 161. 
```
func (s *server) process(stream Stream, reqCh <-chan *discovery.DiscoveryRequest, defaultTypeURL string) error {
...
	knownResourceNames := map[string]map[string]struct{}{}
```

Reply:
Right I saw that, I only ask cause there may be a case when a consumer of this API only uses the cache and writes their own server implementation. If they don't initialize knownResourceNames I believe this will panic. Maybe add a nil check at the top?

Reply:
I see. It should not panic and it doesn't panic right now in Unit tests :). I added this check to avoid adding more changes in all unit tests. Majority of Unit tests passes `knownResourceNames=nil` . There are specific Unit tests for case when `knownResourceNames != nil` to check the logic.
Anyway I will see how the code is changing when streamState is in play.

Reply:
Great that sounds good